command/debug/debug.go

```
@@ -564,6 +571,72 @@ func (c *cmd) captureMetrics(ctx context.Context) error {
 	return nil
 }
 
+func (c *cmd) captureRaft (ctx context.Context) error {
+	reply, err := c.client.Operator().RaftGetConfiguration(nil)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve raft configuration: %v", err)
+	}
+
+	leaderLastCommitIndex := uint64(0)
+	serverIdLastIndexMap := make(map[string]uint64)
+
+	for _, raftServer := range reply.Servers {
+		serverIdLastIndexMap[raftServer.ID] = raftServer.LastIndex
+	}
+
+	for _, s := range reply.Servers {
+		if s.Leader {
+			lastIndex, ok := serverIdLastIndexMap[s.ID]
+			if ok {
+				leaderLastCommitIndex = lastIndex
+			}
+		}
+	}
+
+	// Format it as a nice table.
+	result := []string{"Node\x1fID\x1fAddress\x1fState\x1fVoter\x1fRaftProtocol\x1fCommit Index\x1fTrails Leader By"}
+	for _, s := range reply.Servers {
+		raftProtocol := s.ProtocolVersion
+
+		if raftProtocol == "" {
+			raftProtocol = "<=1"
+		}
+		state := "follower"
+		if s.Leader {
+			state = "leader"
+		}
+
+		trailsLeaderByText := "-"
+		serverLastIndex, ok := serverIdLastIndexMap[s.ID]
+		if ok {
+			trailsLeaderBy := leaderLastCommitIndex - serverLastIndex
+			trailsLeaderByText = fmt.Sprintf("%d commits", trailsLeaderBy)
+			if s.Leader {
+				trailsLeaderByText = "-"
+			} else if trailsLeaderBy == 1 {
+				trailsLeaderByText = fmt.Sprintf("%d commit", trailsLeaderBy)
+			}
+		}
+		result = append(result, fmt.Sprintf("%s\x1f%s\x1f%s\x1f%s\x1f%v\x1f%s\x1f%v\x1f%s",
+			s.Node, s.ID, s.Address, state, s.Voter, raftProtocol, serverLastIndex, trailsLeaderByText))
+	}
+
+	listpeers, err := columnize.Format(result, &columnize.Config{Delim: string([]byte{0x1f})}), nil
```

This may discard the actual error from `columnize.Format` due to err = nil assignment.
```
listpeers, err := columnize.Format(result, &columnize.Config{Delim: string([]byte{0x1f})})
```